cql3/expr/restrictions.cc

```diff
@@ -95,10 +99,21 @@ void validate_token_relation(const std::vector<const column_definition*> column_
             pk.end(), [](auto* c1, auto& c2) {
                 return c1 == &c2; // same, not "equal".
         })) {
-
+        std::vector<const column_definition*> unique_columns;
+        std::ranges::unique_copy(column_defs, std::back_inserter(unique_columns));
+        if (unique_columns.size() < column_defs.size()) {
+            throw exceptions::invalid_request_exception(
+                "The token() function contains duplicate partition key components");
+        }
+        if (unique_columns.size() < pk.size()) {
+            throw exceptions::invalid_request_exception(
+                "The token() function must be applied to all partition key components or none of them");
```

Comment:
To me the part with `none of them` is a bit confusing. The `token()` function can't have empty arguments. I think it would be better to have just the `"The token() function must be applied to all partition key components."`

Reply:
indeed, it was confusing to me when i was reading this error message when testing with Cassandra. but after reading the samples and syntax of `token()` a while, i realized that it was actually quite right, albeit confusing, as you put. and yeah, i was actually experimenting CQL statements including a `token()` call without arguments!

i think the message sent my Cassandra actually means, 

> either you need to use `token(pk, ck)`, where the tuple of `(pk, ck)` is the primary key of the table. or you can use `token(3, 4)`, where *none* of the keys are included, but please don't use `token(pk)`.

but if you still think  

> The token() function must be applied to all partition key components.

is better than Cassandra's error message:

> The token() function must be applied to all partition key components or none of them

i will incorporate the former in the next revision.

Reply:
Ah I didn't realize that it's a message from Cassandra.
Let's leave it as is then, we want to be as compatible with Cassandra as possible.